registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/validator/LostRidRequestValidator.java

```diff
@@ -0,0 +1,205 @@
+
+	/** The Constant REG_STATUS_SERVICE. */
+	private static final String REG_STATUS_SERVICE = "RegStatusService";
+
+	/** The Constant REG_STATUS_APPLICATION_VERSION. */
+	private static final String REG_LOSTRID_APPLICATION_VERSION = "mosip.registration.processor.lostrid.version";
+
+	private static final String REG_LOSTRID_SERVICE_ID = "mosip.registration.processor.lostrid.id";
+
+	/** The env. */
+	@Autowired
+	private Environment env;
+
+	/** The grace period. */
+	@Value("${mosip.registration.processor.grace.period}")
+	private int gracePeriod;
+
+
+	/**
+	 * Validate.
+	 *
+	 * @param registrationStatusRequestDTO the registration status request DTO
+	 * @param serviceId                    the service id
+	 * @throws RegStatusAppException the reg status app exception
+	 */
+	public void validate(LostRidRequestDto lostRidRequestDto)
+			throws RegStatusAppException, WorkFlowSearchException {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(), "",
+				"LostRidRequestValidator::validate()::entry");
+
+		validateId(lostRidRequestDto.getId());
+		validateVersion(lostRidRequestDto.getVersion());
+		validateReqTime(lostRidRequestDto.getRequesttime());
+		validateFilter(lostRidRequestDto.getRequest().getFilters());
+		validateSortField(lostRidRequestDto.getRequest().getSort());
+
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(), "",
+				"LostRidRequestValidator::validate()::exit");
+
+	}
+
+	/**
+	 * Validate id.
+	 *
+	 * @param id the id
+	 * @throws RegStatusAppException the reg status app exception
+	 */
+	private void validateId(String id) throws RegStatusAppException {
+		LostRidValidationException exception = new LostRidValidationException();
+
+		if (Objects.isNull(id)) {
+
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_MISSING_INPUT_PARAMETER_ID, exception);
+		} else if (!env.getProperty(REG_LOSTRID_SERVICE_ID).equals(id)) {
+
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_INVALID_INPUT_PARAMETER_ID, exception);
+
+		}
+	}
+
+
+	/**
+	 * Validate filter.
+	 *
+	 * @param version the version
+	 * @param errors  the errors
+	 * @return true, if successful
+	 * @throws RegStatusAppException
+	 * 
+	 */
+	private void validateFilter(List<FilterInfo> filterInfos) throws WorkFlowSearchException, RegStatusAppException {
+		LostRidValidationException exception = new LostRidValidationException();
+		if (Objects.isNull(filterInfos)) {
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_MISSING_INPUT_PARAMETER_VERSION, exception);
+		}
+		for (FilterInfo filter : filterInfos) {
+			if (filter.getColumnName().equals("name") || filter.getColumnName().equals("email")
+					|| filter.getColumnName().equals("phone") || filter.getColumnName().equals("centerId")
+					|| filter.getColumnName().equals("registrationDate")) {
```

Comment:
I remember there was a requirement to even allow dates on only with max range like say 30 days. Recheck this requirement and add the validation

Reply:
Also validation of filter type is missing

Reply:
taken care, requirement for to narrow down the search result not for date

Reply:
1. There is a empty if condition for filter type validation, please invert the check and throw the exception
2. I remember the requirement of max interval between dates from Sasi, whom did you verify the requirement ?

Reply:
yes but not for interval for record , we r validating the output if more then 5 records is there then we r throwing error, user can define any interval but we only show five record if more then five records is there then we r throwing error

Reply:
Please bring this up in today's evening call and lets have quick discuss with Sasi and we can close this

Reply:
Please don't have a empty if condition. Also make the days 30 as configurable

Reply:
taken care